txpipe/twopoint_fourier.py

```diff
@@ -157,111 +156,128 @@ def load_maps(self):
+            density_fields.append(field)
 
-        d_fields = []
-        wl_fields = []
 
-        if pixel_scheme.name == 'gnomonic':
-            lx = np.radians(pixel_scheme.size_x)
-            ly = np.radians(pixel_scheme.size_y)
-
-            # Apodize the mask
-            if apod_size > 0:
-                if self.rank==0:
-                    print(f"Apodizing mask with size {apod_size} deg and method {apod_type}")
-                mask = nmt.mask_apodization_flat(mask, lx, ly, apod_size, apotype=apod_type)
-            elif self.rank==0:
-                print("Not apodizing mask.")
-
-            for i,d in enumerate(d_maps):
-                # Density for gnomonic maps
-                if self.rank == 0:
-                    print(f"Generating gnomonic density field {i}")
-                field = nmt.NmtFieldFlat(lx, ly, mask, [d], templates=syst_nc) 
-                d_fields.append(field)
-
-            for i,(g1,g2) in enumerate(zip(g1_maps, g2_maps)):
-                # Density for gnomonic maps
-                if self.rank == 0:
-                    print(f"Generating gnomonic lensing field {i}")
-                field = nmt.NmtFieldFlat(lx, ly, mask, [g1,g2], templates=syst_wl) 
-                wl_fields.append(field)
-            
-
-        elif pixel_scheme.name == 'healpix':
-            # Apodize the mask
-            if apod_size > 0:
-                if self.rank==0:
-                    print(f"Apodizing mask with size {apod_size} deg and method {apod_type}")
-                mask = nmt.mask_apodization(mask, apod_size, apotype=apod_type)
-            elif self.rank==0:
-                print("Not apodizing mask.")
-
-
-            for i,d in enumerate(d_maps):
-                # Density for gnomonic maps
-                print(f"Generating healpix density field {i}")
-                field = nmt.NmtField(mask, [d], templates=syst_nc)
-                d_fields.append(field)
-
-            for i,(g1,g2) in enumerate(zip(g1_maps, g2_maps)):
-                # Density for gnomonic maps
-                print(f"Generating healpix lensing field {i}")
-                field = nmt.NmtField(mask, [g1, g2], templates=syst_wl) 
-                wl_fields.append(field)
-
-        else:
-            raise ValueError(f"Pixelization scheme {pixel_scheme.name} not supported by NaMaster")
+        # And then the lensing maps
+        for i, (g1, g2, lw) in enumerate(zip(g1_maps, g2_maps, lensing_weights)):
+            if self.rank == 0:
+                print(f"Generating lensing field {i}")
+            # Same again.  NaMaster knows whether these are spin 0 or spin 2
+            # from the length of the maps list we pass it (1 or 2).
+            if pixel_scheme.name == 'gnomonic':
+                field = nmt.NmtFieldFlat(lx, ly, lw, [g1,g2], templates=syst_wl) 
+            elif pixel_scheme.name == 'healpix':
+                field = nmt.NmtField(lw, [g1, g2], templates=syst_wl) 
```

Comment:
So, just to clarify:
- `lw` contains the sum of weights in each pixel
- `g1/2` contain the weighted mean of gamma_1/2 in each pixel (or zero if the sum of weights in that pixel is zero)
correct?